opus/application/static_media/js/browse.js

```diff
@@ -572,15 +573,25 @@ var o_browse = {
         let contentsView = o_browse.getScrollContainerClass();
         let maxObs = (opus.prefs.view === "cart" ? o_cart.cartCount : opus.resultCount);
 
-        let obsNum = $(`${tab} .thumbnail-container[data-id=${opusId}]`).data("obs") + 1;
+        let obsNum = $(`${tab} .thumbnail-container[data-id=${opusId}]`).data("obs");
         if (obsNum <= maxObs) {
-            let nextElem = $(`${tab} .thumbnail-container[data-obs=${obsNum}]`);
+            let nextElem = (contentsView === ".op-gallery-view" ? $(`${tab} .thumbnail-container[data-obs=${obsNum}]`) : $(`${tab} tr[data-obs=${obsNum}]`));
             if (nextElem.length === 0) {
                 // disable keydown on modal when it's loading
                 // this will make sure we have correct html elements displayed for prev observation
                 $("#galleryViewContents").addClass("op-disabled");
                 opus.prefs[startObsLabel] = obsNum;
                 $(`${tab} ${contentsView}`).infiniteScroll("loadNextPage");
+            } else {
+                // make sure the current element that the modal is displaying is viewable; if not, scroll til it is
+                let scope = $(`${tab} ${contentsView}`);
+                if (!nextElem.isOnScreen($(`${tab} .gallery-contents`))) {
+                    let bottom = scope.offset().top + scope.height();
+                    let elemHeight = nextElem.height();
+                    let offset = Math.max(nextElem.offset().top + elemHeight - bottom, elemHeight);
```

Comment:
This math only works if the elem is off the bottom of the screen, not off the top.